app/platform/fabric/sync/SyncService.ts

```diff
@@ -375,25 +383,63 @@ export class SyncServices {
 		this.synchInProcess.push(synch_key);
 
 		const channel_genesis_hash = client.getChannelGenHash(channel_name);
-		const blockHeight = parseInt(channelInfo.height.low) - 1;
-		// Query missing blocks from DB
-		const results = await this.persistence
+		const latestBlockHeight = parseInt(channelInfo.height.low) - 1;
+
+		// Get the value of noOfBlocks from configuration file
+		const noOfBlocks = network_configs[network_id].noOfBlocks;
+
+		// Calculate the starting block height for sync
+		const startingBlockHeight = Math.max(0, latestBlockHeight - noOfBlocks + 1);
+
+		// Syncing Details
+		logger.info(
+			`Syncing blocks from ${startingBlockHeight} to ${latestBlockHeight}`
+		);
+
+		// Load the latest blocks as per the configuration
+		for (
+			let blockHeight = latestBlockHeight;
+			blockHeight >= startingBlockHeight;
+			blockHeight--
+		) {
+			try {
+				const block = await client.fabricGateway.queryBlockByNumber(
+					channel_name,
+					blockHeight
+				);
+				if (block) {
+					await this.processBlockEvent(client, block, noDiscovery);
```

Comment:
Before calling processBlockEvent, can you please check block is available in db. If not you can add.

Reply:
addressed this by adding a conditional check in the code. Now, before calling processBlockEvent, we verify whether the block is available in the database. If the block is not present, we add it as needed. This check ensures that processBlockEvent is only executed when the required data is available in the database, helping to maintain the integrity as expected.